core/api/src/services/openai/assistant.ts

```diff
@@ -134,30 +135,36 @@ export const Assistant = (): ChatAssistant => {
     }
   }
 
-  const processAction = async (run: OpenAI.Beta.Threads.Runs.Run) => {
+  const processAction = async (run: OpenAI.Beta.Threads.Runs.Run): Promise<string[]> => {
     const action = run.required_action
     assert(action?.type === "submit_tool_outputs")
 
-    const name = action.submit_tool_outputs.tool_calls[0].function.name
-    assert(name === "queryBlinkKnowledgeBase")
+    const outputs: string[] = []
 
-    const args = action.submit_tool_outputs.tool_calls[0].function.arguments
-    const query = JSON.parse(args).query_str
+    for (const toolCall of action.submit_tool_outputs.tool_calls) {
+      const name = toolCall.function.name
+      assert(name === "queryBlinkKnowledgeBase")
```

Comment:
not a blocker but this is not a standard way to throw an exception. if we want to use it please catch properly `AssertionError` in the catch block